### PR TITLE
Wait for PV to delete in finalizer of snapshot_restore_factory and pvc_clone_factory

### PR DESCRIPTION
**Pull request by jilju**
Wait for PV to delete in finalizer of snapshot_restore_factory and pvc_clone_factory

Signed-off-by: Jilju Joy <jijoy@redhat.com>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -3120,10 +3137,27 @@ def finalizer():
...
+        pv_objs = []
+
+        # Get PV form PVC instances and delete PVCs
         for instance in instances:
             if not instance.is_deleted:
+                pv_objs.append(instance.backed_pv_obj)
                 instance.delete()
                 instance.ocp.wait_for_delete(instance.name)
 
+        # Wait for PVs to delete
```
Comment (ebenahar) [inline tests/conftest.py:3149]: this code is a dup of https://github.com/red-hat-storage/ocs-ci/pull/3215/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R2961-R2973

Please create a function for it

Comment (jilju) [reply tests/conftest.py:3167]: Created helpers.wait_for_pv_delete function to handle this.

**Review by wusui — APPROVED** [review]
Looks good.  I assume that tests/conftest.py call around line 2969 is the other place where the duplicated code that ebenahar commented on existed.

File: tests/conftest.py (hunk 1 above)
Comment (ebenahar) [reply tests/conftest.py:3167]: thanks!

**Review by ebenahar — APPROVED** [review]

**Pull request merged**